components/support/find-places-db/src/lib.rs

```diff
@@ -40,23 +40,29 @@ impl PlacesLocation {
 }
 
 pub fn get_all_places_dbs() -> Result<Vec<PlacesLocation>> {
-    let mut path = match dirs::home_dir() {
-        Some(dir) => dir,
-        None => bail!("No home directory found!"),
-    };
-    if cfg!(windows) {
+    let path = if cfg!(windows) {
+        let mut path =
+            dirs::home_dir().ok_or_else(|| anyhow::anyhow!("No home directory found!"))?;
         path.extend(&["AppData", "Roaming", "Mozilla", "Firefox", "Profiles"]);
+        path
     } else {
         let out = String::from_utf8(process::Command::new("uname").args(["-s"]).output()?.stdout)?;
         info!("Uname says: {:?}", out);
         if out.trim() == "Darwin" {
+            let mut path =
+                dirs::home_dir().ok_or_else(|| anyhow::anyhow!("No home directory found!"))?;
             // ~/Library/Application Support/Firefox/Profiles
             path.extend(&["Library", "Application Support", "Firefox", "Profiles"]);
+            path
         } else {
             // I'm not actually sure if this is true for all non-macos unix likes.
+            // Using `dirs::config_dir()` gives us XDG support.
+            let mut path =
+                dirs::config_dir().ok_or_else(|| anyhow::anyhow!("No config directory found!"))?;
```

Comment:
So here we end up with `$XDG_CONFIG_HOME/.mozilla/firefox` ? That's incorrect, it should be `$XDG_CONFIG_HOME/mozilla/firefox`

Reply:
huh - I was taking this from https://searchfox.org/firefox-main/rev/60308bc3792ef201b82377682de068a5a1c72575/toolkit/crashreporter/crash_helper_server/src/logging/env.rs#38-42 which also seems to use `.mozilla` in that case. That other code also has a strange `||` which seems odd - it will use the home directory if it exists even if `MOZ_LEGACY_HOME` isn't set - I didn't copy those semantics.

I'll update this patch as you note though, thanks!

Reply:
Looks like it's my mistake :(